File: frontend/types/history.ts
export type ProjectHistoryAction =
  | "project_created"
  | "project_updated"
  | "scenario_created"
  | "scenario_updated"
  | "report_generated";

export type ProjectHistoryEvent = {
  action: ProjectHistoryAction;
  actor: string;
  occurred_at: string;
  summary: string;
};

File: backend/app/schemas/history.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

AuditEventAction = Literal[
    "project_created",
    "project_updated",
    "scenario_created",
    "scenario_updated",
    "report_generated",
]


class ProjectHistoryEventResponse(BaseModel):
    action: AuditEventAction
    actor: str
    occurred_at: datetime
    summary: str
